src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_SwitchExpression.cs

```diff
@@ -92,10 +92,7 @@ private BoundExpression LowerSwitchExpression(BoundConvertedSwitchExpression nod
                     sectionBuilder.Add(_factory.Label(arm.Label));
                     var loweredValue = _localRewriter.VisitExpression(arm.Value);
                     if (GenerateSequencePoints)
-                    {
-                        // Should go through this._localRewriter._instrumenter; see https://github.com/dotnet/roslyn/issues/42810
-                        loweredValue = new BoundSequencePointExpression(arm.Value.Syntax, loweredValue, loweredValue.Type);
-                    }
+                        loweredValue = this._localRewriter._instrumenter.InstrumentSwitchExpressionArmExpression(arm.Value, loweredValue, _factory);
```

Comment:
>loweredValue = this._localRewriter._instrumenter.InstrumentSwitchExpressionArmExpression(arm.Value, loweredValue, _factory); [](start = 24, length = 124)

It is not clear why instrumentation would be linked to generation of sequence points.

Reply:
If you prefer I will rename `GenerateSequencePoints` to `GenerateInstrumentation`.